packages/atomic/cypress/integration/facets/category-facet-assertions.ts

```diff
@@ -0,0 +1,256 @@
+import {doSortAlphanumeric, doSortOccurences} from '../../utils/componentUtils';
+import {RouteAlias} from '../../utils/setupComponent';
+import {ComponentErrorSelectors} from '../component-error-selectors';
+import {ResultListSelectors} from '../result-list-selectors';
+import {
+  hierarchicalField,
+  CategoryFacetSelectors,
+  categoryFacetComponent,
+  BreadcrumbSelectors,
+} from './category-facet-selectors';
+
+// TODO: move util out
+function should(should: boolean) {
```

Comment:
I feel functions like this one are starting to enter the realm of over-engineering?

Personally, I would keep the `it` blocks and their descriptions explicitly written out in the test files. This would provide flexibility to express nuance (examples, edge cases) in the descriptions. It would also avoid having utility `should` functions.

The actual assertions can remain in this file though, reused inside multiple `it` blocks.

Reply:
I chuckled when writing this method, but it prevented me from rewriting the descriptions, which is one of the main purpose of this PR: to minimize repetition sand maximize consistency & reusability of assertions. 

Having the `it` description & the assertions in 2 different places would require checking back and forth to make sure the "assertion code" fits with the description, here, they go together.

I rather use a trivial util method than manually writing it out every time or copy-pasting. For example I would have to write out `should (or not) display a "All Categories" button` 7 times in this test suite only. I find Cypress tests have way more repetitions than when writing UTs, conditions have to be checked and rechecked many times. 

> provide flexibility to express nuance (examples, edge cases)

Most of the context & nuances are expressed using the describe blocks, which prepare the interface for the assertions. This pattern kind of enforces the good practice of writing 1 check/expect/assert per "it" block. Then as many assert methods as needed can then be written out, along with their it block.

Reply:
I'm fine if it "helps reduce some repetition", but, yeah.... repetition is not always bad, you know ;) 

It's boring to write, sure, but it's easier to read.

Anyway, as long as we don't "force this" for all tests, if it makes sense for a particular test suite 👍